src/EventTickets/Hooks/EventTicketsAdminPage.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Give\EventTickets\Hooks;
+
+use Give\EventTickets\ListTable\EventTicketsListTable;
+use Give\Helpers\EnqueueScript;
+
+class EventTicketsAdminPage
+{
+    /**
+     * @var string
+     */
+    private $apiRoot;
+
+    /**
+     * @var string
+     */
+    private $apiNonce;
+
+    /**
+     * @var string
+     */
+    private $adminUrl;
+
+    public function __construct()
+    {
+        $this->apiRoot = esc_url_raw(rest_url('give-api/v2/admin/event-tickets'));
+        $this->apiNonce = wp_create_nonce('wp_rest');
+        $this->adminUrl = admin_url();
+    }
+
+    /**
+     * @unreleased
+     */
+    public function registerMenuItem()
+    {
+        add_submenu_page(
+            'edit.php?post_type=give_forms',
+            esc_html__('Events', 'give'),
+            esc_html__('Events', 'give') . ' <span class="give-menu-badge">Beta</span>',
+            'edit_give_forms',
+            'give-event-tickets',
+            [$this, 'render']
+        );
+    }
+
+    /**
+     * @unreleased
+     */
+    public function loadScripts()
+    {
+        $data = [
```

Comment:
What do you think about extracting this data array to a `DataTransferObject`?

Reply:
This is specific to this use and has been converted into its own action. I don't think it's necessary to change it but let me know.